add mfpca.face

ecui1 opened the pull request:
We added the mfpca.face() function to refund, which implements the fast MFPCA method that we have been working on. We checked the package and it worked well on our laptops and passed R CMD check.

Commit:
add mfpca.face

fabian-s commented:
Thank you for the pull request, that's great.
 
Please add some unit tests for the new code that uses its basic functionality on some simple examples. 

This also gives us a new dependency on `simex`,  for using just a single function. Would it be possible to avoid `simex::diag.block` and use a straight kronecker product or `Matrix::bdiag`? We already depend on `Matrix` anyway, so either of those would be preferable.

fabian-s commented on R/mfpca.face.R:
```diff
@@ -0,0 +1,482 @@
+    # m: orders of difference penalty, with default values 2
+    # knots.option: type of knots placement, with default values "equally-spaced"
+    
+    #require(splines)
```
please remove unnecessary, commented out code

ecui1 replied:
The commented code is removed now.

fabian-s commented on R/mfpca.face.R:
```diff
@@ -0,0 +1,482 @@
+      eSig = eigen(Sig)
+      V = eSig$vectors
+      E = eSig$values
+      if(min(E)<=0.0000001) {#cat("Warning! t(B)%*%B is singular!\n");
```
see above -- plz remove or re-activate these warnings (and use `warning` or `message`, not `cat`)

ecui1 replied:
These warning messages are removed now from the code.

Commit:
add mfpca.face with test examples

ecui1 commented:
Thanks for the review. We have updated the function and pushed it to the forked repo.

We have added some unit tests to 'test-mfpca.R' in the tests folder, and removed the dependency on simex package. 

The new function works well, and the package passed R CMD check.

Please let us know if you have any other comments/questions.

fabian-s reviewed (approved):
Thanks, that was quick!

fabian-s commented:
I'll let @julia-wrobel  take care of the rest ... ;)

Commit:
remove commented code

julia-wrobel commented:
Thanks for reviewing @fabian-s!
